### PR TITLE
test: CodeFixVerifier: verify the change fixed the diagnostic

### PR DESCRIPTION
**Pull request by bcollamore**
Currently, the CodeFixer compares the new code with the old code. However, it didn't actually confirm that it fixed the diagnostic! Added support for that, and then cleaned up a few tests that had residual diagnostics that now fired (causing the test to fail).

### CODE REVIEW

File: Philips.CodeAnalysis.Test/Verifiers/CodeFixVerifier.cs [hunk 1]
```diff
@@ -113,6 +115,9 @@ private void VerifyFix(DiagnosticAnalyzer analyzer, CodeFixProvider codeFixProvi
 				}
 			}
 
+			//after applying all of the code fixes, there shouldn't be any problems remaining
+			Assert.IsTrue(allowNewCompilerDiagnostics || !analyzerDiagnostics.Any(), $@"After applying the fix, there still exists {analyzerDiagnostics.Length} diagnostic(s), including {analyzerDiagnostics.FirstOrDefault()?.Id}");
```
Comment (ynsehoornenborg) [inline Philips.CodeAnalysis.Test/Verifiers/CodeFixVerifier.cs:119]: Replace the `analyzerDiagnostics.FirstOfDefault()?.Id` with `string.Join(",", analyzerDiagnostics.Select(diag => diag.Id)` to make sure we show all the diagnostics

**Review by ynsehoornenborg — APPROVED** [review]

File: Philips.CodeAnalysis.Test/Verifiers/CodeFixVerifier.cs (hunk 1 above)
Comment (bcollamore) [reply Philips.CodeAnalysis.Test/Verifiers/CodeFixVerifier.cs]: done

Comment (ynsehoornenborg) [reply Philips.CodeAnalysis.Test/Verifiers/CodeFixVerifier.cs]: I don't see the code change

Comment (ynsehoornenborg) [reply Philips.CodeAnalysis.Test/Verifiers/CodeFixVerifier.cs]: Never mind, GitHub played tricks on me

**Review by ynsehoornenborg — APPROVED** [review]

**Pull request merged**